### PR TITLE
PLAT-1798: Table: Move header updating to zen-table.tsx

### PR DESCRIPTION
**Pull request by chavlji**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1798)

2st part of ticket [PLAT-1667: Table: indeterminate state inconsistent](https://reciprocitylabs.atlassian.net/browse/PLAT-1667)

#### Description
As per JIRA ticket

### CODE REVIEW

**Review by mihaelpalfi — APPROVED** [review]
Fantasctic! Much cleaner code. Tested it. I noticed only one thing, nothing mayor : 
NIT : Shouldn't the columns be aligned with header column?
http://localhost:6006/?path=/docs/lists-table-table--with-fragments
image

**Comment by chavlji** [comment]
@mihaelpalfi Yeah, I've noticed it too and fixed it in the following PR's commit:
https://github.com/reciprocity/zen-ui/pull/310/commits/f63da644d615f581fbf377d96cbc99d5de77f1a0

**Review by f-jauregui — APPROVED** [review]
Excellent work, Jan! Working as expected 👍

File: src/components/zen-table/zen-table-helpers.tsx [hunk 1]
```diff
@@ -64,22 +78,34 @@ export const cleanupTableStructure = function (table: HTMLZenTableElement): void
     }
   };
 
+  const updateHeaderSelectCheckbox = (header, rows) => {
+    const allSelected = rows.length && rows.every(n => n.selected);
+    const someSelected = rows.some(n => n.selected);
```
Comment (homer0) [inline src/components/zen-table/zen-table-helpers.tsx:83]:
```suggestion
    const someSelected = !allSelected && rows.some(n => n.selected);
```

There's no need to do the `.some` if the `.every` returned `true`.

**Review by homer0 — APPROVED** [review]
🤘 Tiny comment :P

**Pull request merged**